polyfill/intersection-observer-test.js

```diff
@@ -941,6 +941,652 @@ describe('IntersectionObserver', function() {
 
   });
 
+  describe('same-origin iframe', function() {
+    var iframe, win, doc;
+    var iframeTargetEl1, iframeTargetEl2;
+
+    beforeEach(function() {
+      iframe = document.createElement('iframe');
+      iframe.setAttribute('frameborder', '0');
+      iframe.setAttribute('scrolling', 'yes');
+      iframe.style.position = 'fixed';
+      iframe.style.top = '0px';
+      iframe.style.width = '100px';
+      iframe.style.height = '200px';
+      iframe.srcdoc =
+        '<!DOCTYPE html><html><body>' +
+        '<style>' +
+        'body {margin: 0}' +
+        ' .target {height: 200px; margin-bottom: 2px; background: blue;}' +
+        '</style>';
+      return new Promise(function(resolve, reject) {
```

Comment:
When this polyfill was written (years ago) it was designed to work all the way back to IE7.

Using `Promise` here would break the tests when run in IE, but I think you could get away with still using it by adding that polyfill to the polyfills the test runner loads: https://github.com/w3c/IntersectionObserver/blob/master/polyfill/intersection-observer-test.html#L22

Reply:
Good point. I removed the promise. The rest of tests are using `done` callback, so this one should do as well.